UnitedIncome/components
# [APP-2800] Adding Percy pull request check

### Description
> Please provide a brief description of your changes below.

This change checks to see if visual regression tests are needed. With these changes, when CircleCI builds `develop` it will test to see if an active pull request is open against the `master` branch. If it is it will pass the check and run the tests, if it is not it will still greenlight the build, but not run the Percy checks.

With these changes we'll be able to test to make sure that we are only running Percy before we attempt a release, which means we'll no longer have a new version cut before we see the visual changes.

### Testing Instructions
> What should be done in order to test this pull request, or re-create its effects.

* We'll need to try this in CircleCI, I don't know if there's a way to test these locally. You can test the bash scripts locally that I've put into the CircleCI though, you'll need an access token first with public repo read access saved as an env variable.

### Comments
> Any additional comments that will aid with the review process.

The bash script in the Circle config file wraps the node command. The reason for this is because CircleCI handles exit statuses of 1 as a build failure. With the wrapper the yarn command can exit and cease to run, but it won't fail the build. Technically this workflow will run on every commit to develop, it just won't run the visual tests on every commit unless a pull request is open. 

- [JIRA Ticket](https://unitedincome.atlassian.net/browse/app-2800)
